contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+      require(!fulfillments[_fulfillmentId].accepted);
+      _;
+  }
+
+  modifier sameLength(uint l1, uint l2){
+      require(l1 == l2);
+      _;
+  }
+
+  modifier sumToOne(uint[] _numerators, uint _denomenator){
+      uint sum = 0;
+      for (uint i = 0; i < _numerators.length; i++){
+        sum += _numerators[i];
+      }
+      require(sum == _denomenator);
+      _;
+  }
+
+
+  /*
+   * Public functions
+   */
+
+  function ()
+  public
+  payable
+  {
+
+  }
+
+  function initializeBounty(
+      address _controller,
+      string _data)
+      public
+  {
+    require(controller == address(0));
+    require(_controller != address(0));
+    // an controller of a bounty is only 0x0 when it is uninitialized,
+    // so this check prevents initialization from being called multiple times
+
+    controller = _controller;
+    data = _data;
+
+  }
+
+  function refundableContribute(uint[] _amounts, StandardToken[] _tokens)
+  public
+  payable
+  {
+      require(_amounts.length == _tokens.length);
+
+      contributions.push(Contribution(msg.sender, _amounts, _tokens, false));
+      uint contributionId = contributions.length - 1;
+      for (uint i = 0; i < _amounts.length; i++){
+          require(!contributions[contributionId].hasContributed[_tokens[i]]);
+          contributions[contributionId].hasContributed[_tokens[i]] = true;
+          if (_tokens[i] == address(0)){
+              require(msg.value == _amounts[i]);
+          } else {
+              require(_tokens[i].transferFrom(msg.sender, this, _amounts[i]));
+          }
+      }
+  }
+
+  function refundContribution(uint _contributionId)
+  public
+  hasNotPaid
+  onlyContributor(_contributionId)
+  notYetRefunded(_contributionId)
+  {
+      contributions[_contributionId].refunded = true;
+      for (uint i = 0; i < contributions[_contributionId].amounts.length; i++){
+          if (contributions[_contributionId].tokens[i] == address(0)){
+              contributions[_contributionId].contributor.transfer(contributions[_contributionId].amounts[i]);
+          } else {
+              require(contributions[_contributionId].tokens[i].transfer(contributions[_contributionId].contributor, contributions[_contributionId].amounts[i]));
+          }
+      }
+  }
+
+  function fulfillBounty(address[] _fulfillers, uint[] _numerators, uint _denomenator, string _data)
+      public
+      validateNotTooManyFulfillments
+      sameLength(_fulfillers.length, _numerators.length)
+      sumToOne(_numerators, _denomenator)
+  {
+      fulfillments.push(Fulfillment(_fulfillers, _numerators, _denomenator, _data, false));
+
+      BountyFulfilled((fulfillments.length - 1));
+  }
+
+  function calculateFraction(uint _balance, uint _numerator, uint _denomenator)
+      public
+      pure
+      returns (uint newBalanceDiv)
+  {
+    require(_denomenator != 0);
+
+    //first multiplies by the numerator
+    uint newBalanceMult = _balance * _numerator;
```

Comment:
ahh i see - setting it to pure is forcing newBalancMult as memory vs. storage.

Reply:
hmmm - no that would default to the stack. why are you setting this function to pure?